test/it-tests.js

```diff
@@ -30,11 +30,105 @@ var clusterSizes = [1, 2, 3, 4, 5, 6, 7, 10, 21, 25, 30];
 // Global counter to record how many tests have failed.
 var testFailures = 0;
 
+var features = {
+    'join': {
+        mandatory: true,
+        tests: [
+            './join-tests'
+        ]
+    },
+    'ping': {
+        mandatory: true,
+        tests: [
+            './ping-tests'
+        ]
+    },
+    'ping-req':{
+        mandatory: true,
+        tests: [
+            './ping-req-tests'
+        ]
+    },
+    'reincarnate': {
+        mandatory: true,
+        tests: [
+            './incarnation-no-tests'
+        ]
+    },
+    'gossip': {
+        mandatory: true,
+        tests: [
+            './piggyback-tests'
+        ]
+    },
+    'admin': {
+        mandatory: true,
+        tests: [
+            './admin-tests'
+        ]
+    },
+
+    // features implemented in only one language
+    'reaping-faulty-nodes': {
+        tests: [
+            './reaping-faulty-nodes'
+        ]
+    }
+};
+
+function selectTests(options) {
+    var select;
+
+    var only = options.only || [];
+    var selectedFeatures = options.features || [];
+
+    if (only.length > 0) {
+        select = function (obj, feature) {
+            // add selected features
+            if (only.indexOf(feature) >= 0) {
+                return true;
+            }
+        };
+    } else {
+        select = function (obj, feature) {
+            // always run all mandatory features
+            if (obj.mandatory === true) {
+                return true;
+            }
+
+            // add selected features
+            if (selectedFeatures.indexOf(feature) >= 0) {
+                return true;
+            }
+
+            // drop other tests
+            return false;
+        };
+    }
+
+    return _
+        .chain(features)
+        .filter(select)
+        .map(function (obj) {
+            return obj.tests
+        })
+        .flatten()
+        .value();
+}
+
+// collect is a commander helper function
+function collect(val, memo) {
+  memo.push(val);
+  return memo;
+}
+
 function main() {
     program
         .version(require('../package.json').version)
         .option('-s, --sizes <clusterSizes>', 'Cluster sizes to test against. Default: \'' +
              JSON.stringify(clusterSizes) + '\'')
+        .option('--enable-feature <feature>', 'Run tests for experimental features', collect, [])
```

Comment:
Solved via printing `#: WARNING skipping test suite: ......` and fix the passthrough in `tap-filter`